src/platform/plugins/shared/controls/public/controls/data_controls/range_slider/components/range_slider_control.tsx

```diff
@@ -251,7 +251,16 @@ export const RangeSliderControl: FC<Props> = ({
         minInputProps={minInputProps}
         maxInputProps={maxInputProps}
         value={[displayedValue[0] || displayedMin, displayedValue[1] || displayedMax]}
-        onChange={([minSelection, maxSelection]: [number | string, number | string]) => {
+        onChange={([minSelection, maxSelection]: [number | string, number | string], _, ev) => {
+          const originatingInput = ev?.currentTarget.getAttribute('data-test-subj');
+
+          if (originatingInput === 'rangeSlider__lowerBoundFieldNumber') {
+            // preserve original upper bound selection if only lower bound number field changed
+            maxSelection = displayedValue[1];
+          } else if (originatingInput === 'rangeSlider__upperBoundFieldNumber') {
+            // preserve original lower bound selection if only upper bound number field changed
+            minSelection = displayedValue[0];
+          }
```

Comment:
I don't love using `data-test-subj` in this way... It feels like it could very easily break, since this attribute should be reserved for tests and, if it were to change down the line, we could easily forget to update this logic. But I can't quite figure out how to handle this better, since EUI doesn't differentiate between min and max changes in a more meaningful way.

Let me stew on this a bit 🤔

Reply:
I switched to using unique IDs instead of test subjects in b6a99559fe1befa1c1a22804782d36a4b923ba9b.

Reply:
I was hoping we could provide separate `onChange` handlers to each input - and we can! But the generic `onChange` for the slider still fires regardless of where the input comes from (i.e. both the input `onChange` and the slider `onChange` fire when typing a value; only the slider `onChange` fires when dragging). There is no valid distinction between the two events, so we have no way of saying "don't fire the slider `onChange` if the event is coming from the input"

Unfortunately, I think this means we have to go with the current approach. It was worth a shot! :P